### PR TITLE
chore(docs): Docs revamp, update folder structure Boilerplate/*

### PR DESCRIPTION
**Pull request by jamonholmgren**
This is an effort to improve the docs measurably, both in structure and filling in gaps of content.

It will break some documentation links, but since docs.infinite.red is new, I'd rather do it now before there are too many external links.

It's not yet ready to review but feel free to add your thoughts anytime.

image

Main changes:

* Updated Boilerplate.md to be more of an entry point to discussing the folder structure (and boilerplate in general), rather than immediately jumping into folder structure discussion.
* Created the full boilerplate folder tree in ./Boilerplate/*, such as ./Boilerplate/app/app.tsx.md. Each of these new docs discusses the purpose of the file and how it is used. It's a starting point for further expansion.
* Updated the Boilerplate.md file to link to the new files in ./Boilerplate/*.
* Moved ./Components into ./Boilerplate/components. This will probably break some external links.
* Moved ./Theming as well.
* Fixed internal links to the new locations.
* Created an Expo section with Expo Go, CNG, and DIY pages. Also an EAS page.
* Moved Expo-and-Ignite into the new Expo section.
* Updated Styling a bit
* Updated the docs/README.md to link to all the new docs

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -2,70 +2,95 @@
...
+If you're new to Ignite, check out these resources:
 
 - Check out the [Getting Started with Ignite](https://www.youtube.com/watch?v=KOSvDlFyg20) video for a 13 minute overview!
-- Jamon's Code Quest on [Ignite 9](https://www.youtube.com/watch?v=QmkMsUYrTlk)
+- Jamon's Code Quest on [What's new in Ignite 9](https://www.youtube.com/watch?v=QmkMsUYrTlk)
+- Check out our [Guide](./Guide.md) for a walkthrough of the boilerplate project and how to use it
+
+<table>
+<tr>
+<td><img alt="ignite-01-sign-in" src="https://github.com/infinitered/ir-docs/assets/1479215/d84d5efa-363d-44bc-a7b3-6ec70d878ae3" /></td>
```
Comment (Jpoliachik) [inline docs/README.md:36]: where are these images? Im curious - I don't see them in either ir-docs or ignite repo, so I'm curious how they end up hosted here!

File: docs/concept/Folder-Structure.md [hunk 2]
```diff
@@ -57,19 +57,19 @@ app
 
 This is where your components will live, the reusable building blocks to create your screens. A handful of built-in components come with Ignite that are adaptable to any custom design system you wish to implement. Below are links to further documentation about each component:
```
Comment (Jpoliachik) [inline docs/concept/Folder-Structure.md:58]: this Folder Structure page might not be necessary, a lot of the info here is duplicated in Boilerplate.md - so we probably don't want to maintain both

**Review by Jpoliachik — COMMENTED** [review]
I dig it! 
Definitely nice to have the file structure mirror the project

**Review by Jpoliachik — APPROVED** [review]

**Review by frankcalise — APPROVED** [review]
I like this idea a lot! Well done!

**Comment by markrickert** [comment]
haha, these changes are going to conflict with mine in #2627. I'll take care of it.

**Pull request merged**